### PR TITLE
🔧 Add continuous integration

### PR DESCRIPTION
**Pull request by liberaliscomputing**
This PR adds a GitHub action workflow for continuous integration of IG validation.

### CODE REVIEW

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Continuous Integration
+
+on: [push, pull_request]
+
+jobs:
+  validate:
+    name: Validate Impletementation Guide
```
Comment (fiendish) [inline .github/workflows/continuous_integration.yml:7]:
```suggestion
    name: Validate Implementation Guide
```

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install SUSHI and add FSH definitions
+        run: |
+          npm install -g fsh-sushi
+          sushi .
+
+      - name: Update IG publisher
+        run: |
+          sudo apt-get update
```
Comment (fiendish) [inline .github/workflows/continuous_integration.yml:39]: not that it matters, but update was already run in the jekyll step

File: .github/workflows/continuous_integration.yml (hunk 1 above)
Comment (liberaliscomputing) [reply .github/workflows/continuous_integration.yml]: Modified!

File: .github/workflows/continuous_integration.yml (hunk 2 above)
Comment (liberaliscomputing) [reply .github/workflows/continuous_integration.yml]: Removed!

**Review by torstees — APPROVED** [review]
Looks fine to me.

**Review by znatty22 — COMMENTED** [review]
Meen I'm not too familiar with GH workflows but will the image layers get cached? It looks like your job installs the dependencies for the IG publisher on each run. If the image layers aren't cached seems like each job execution would take a long time? Does GH workflows allow you to supply your own image with the dependencies built in?

**Comment by liberaliscomputing** [comment]
> Meen I'm not too familiar with GH workflows but will the image layers get cached? It looks like your job installs the dependencies for the IG publisher on each run. If the image layers aren't cached seems like each job execution would take a long time? Does GH workflows allow you to supply your own image with the dependencies built in?

@znatty22, yeah, actually building OpenJDK and Jekyll every time is a huge bottleneck. As far as I figured so far, however, only public images can be used as GitHub containers and private images are not supported. It might've got changed and let me took a further look.

**Review by bwalsh — APPROVED** [review]
Looks good.  I double checked that `./_genonce.sh`  will set the $? variable.  

One suggestion for a future version  might be to just run sushi to validate `*.fsh` files.

**Pull request merged**